src/services/events/reducer.js

```diff
@@ -1,17 +1,36 @@
 import {
-  UPDATE_EVENTS,
+  RECIEVE_EVENTS,
+  DELETE_EVENT_SUCCESSFUL,
+  CREATE_EVENT_SUCCESSFUL,
+  UPDATE_EVENT_SUCCESSFUL,
 } from './actions';
 
 const initialState = {
-  school: null,
-  gender: null,
-  gradYear: null,
+  events: [],
 };
 
 const reducer = (state = initialState, action) => {
   switch (action.type) {
-    case UPDATE_EVENTS:
-      return Object.assign({}, state, { school: action.school, gender: action.gender, gradYear: action.gradYear });
+    case RECIEVE_EVENTS:
+      return Object.assign({}, state, { events: action.events });
+    case DELETE_EVENT_SUCCESSFUL:
+      const newDeleteEvents = state.events.filter((event) => (
+        event.id !== action.id
+      ));
+      return Object.assign({}, state, { events: newDeleteEvents });
+    case CREATE_EVENT_SUCCESSFUL:
+      return Object.assign({}, state, { events: [...state.events, Object.assign({}, action.event)] });
+    case UPDATE_EVENT_SUCCESSFUL:
+      // Deep copy events array, or redux won't update correctly
+      const newUpdateEvents = Array.from(state.events);
+      for(let i = 0; i < newUpdateEvents.length; i++) {
```

Comment:
Use the `map` function here. Return original if id doesn't match, return updated if id does